### PR TITLE
Konstantinos / Bug - Fix Typescript errors in deriv.com

### PR DESCRIPTION
**Pull request by konstantinosG-deriv**
Changes:

-   ...

## Type of change

-   [x] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/common/validation.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ const validation = {
             return null
         }
     },
-    required: (input, message) => {
+    required: (input, message?: any) => {
```
Comment (niloo-fs) [inline src/common/validation.ts:26]: Base on current master you don't need to define `any` type for message here!

File: src/common/validation.ts [hunk 2]
```diff
@@ -44,7 +44,7 @@ const validation = {
             return null
         }
     },
-    alphabetic: (input, message) => {
+    alphabetic: (input, message?: any) => {
```
Comment (niloo-fs) [inline src/common/validation.ts:47]: Same here

**Review by niloo-fs — APPROVED** [review]
LGTM

**Review by niloo-fs — APPROVED** [review]

File: src/components/containers/flex.ts [hunk 3]
```diff
@@ -23,6 +23,8 @@ type FlexType = {
     tablet_ai?: string
     tablet_jc?: string
     tablet_fw?: string
+    scroll?: boolean | string
+    margin?: string
```
Comment (mitra-deriv) [inline src/components/containers/flex.ts:27]: It seems like they are unused and can be removed, Would you please check 🙏

File: src/components/containers/container.ts [hunk 4]
```diff
@@ -14,6 +14,7 @@ type ContainerType = {
     fd?: string
     wrap?: string
     fw?: string
+    scroll?: boolean
```
Comment (mitra-deriv) [inline src/components/containers/container.ts:17]: same here 🙏

**Comment by niloo-fs** [comment]
@konstantinosG-deriv Could u pls resolve the conflicts

File: src/common/validation.ts [hunk 5]
```diff
@@ -29,7 +29,7 @@ const validation: TValidationType = {
             return null
         }
     },
-    required: (input, message) => {
+    required: (input, message?: string) => {
```
[the remark this answers is not in the document]
Comment (konstantinosG-deriv) [reply src/common/validation.ts:32]: Looks like it was put there by @mohammad a few weeks ago, didn't notice. Can remove that, thanks 👍

File: src/components/containers/flex.ts (hunk 3 above)
Comment (konstantinosG-deriv) [reply src/components/containers/flex.ts:27]: They are used in the file src/templates/article.tsx 
In these components: 
**<StickyBreadCrumbsWrapper scroll={visible}>**
** <Flex fd="column" margin="0 auto" ai="center">**

File: src/components/containers/container.ts (hunk 4 above)
Comment (konstantinosG-deriv) [reply src/components/containers/container.ts:17]: Used in the Component: BreadcrumbsWrapper

File: src/components/layout/livechat.tsx [hunk 6]
```diff
@@ -78,7 +78,7 @@ const LiveChat = ({ is_banner_shown }: LiveChatProps) => {
                     is_eu_country={is_eu_country}
                     is_rtl={is_rtl}
                     onClick={() => {
-                        LC_API.open_chat_window()
+                        window.LC_API.open_chat_window()
```
[the remark this answers is not in the document]
Comment (konstantinosG-deriv) [reply src/components/layout/livechat.tsx:81]: Oh looks like I changed the wrong file. I was trying to solve the error on file **src/pages/livechat/index.tsx:40** which was `Property 'open_chat_window' does not exist on type '{}'.ts(2339)` . Is it acceptable to use the window object there or do we have any other ideas on how to fix this error? 🙏

[the remark this answers is not in the document]
Comment (niloo-fs) [reply src/components/layout/livechat.tsx:81]: No, please don't use `window` for this,
for fixing this issue you just need to define a type for livechat custom hook

sth like this
```
type TLC_API = {
    open_chat_window?: () => void
}

export const useLivechat = (): [boolean, TLC_API] => {
    const [is_livechat_interactive, setLiveChatInteractive] = useState(false)
    const LC_API = (isBrowser() && window.LC_API) || {}
    const [is_logged_in, setLoggedIn] = useState(false)
```

**Pull request merged**